### ISSUE TITLE
Bug: file_get_contents in XSLT Transformation Breaks jQuery AJAX Data Transport (Development Environment Only)

### ISSUE THREAD

**Issue by ferasshita** [codeigniter4/CodeIgniter4#9475]
### PHP Version

8.3

### CodeIgniter4 Version

4.4.3

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### What happened?

When using file_get_contents to render another CodeIgniter 4 page within an XSLT transformation process, it causes jQuery AJAX requests to fail to transport any data. This issue only occurs in the development environment and does not manifest in production.

### Steps to Reproduce

1. Set up a CodeIgniter 4 project in the development environment.
2. Create an XSLT file and a corresponding PHP script to perform XSLT transformation.
3. Use file_get_contents to render another CodeIgniter 4 page within a callback function.
4. Attempt to make a jQuery AJAX request from the client side.
5. Observe that the AJAX request does not transport any data.
**Code Snippet:**
```php
// Process XSLT transformation
$xsl = new DOMDocument();
$xsl->load(WRITEPATH.'src/Apps/'.$app.'/pages/'.$page.'.xslt');

$proc = new XSLTProcessor();
$proc->importStylesheet($xsl);
$output = $proc->transformToXML(new DOMDocument());

// Replace component tags with templates
foreach ($templateDirs as $component => $dir) {
    $output = preg_replace_callback(
        '/<'.$component.'>(.*?)<\/'.$component.'>/',
        function($matches) use ($dir, $component) {
            echo file_get_contents($dir."/".$matches[1]);
        },
        $output
    );
}

// Final output
echo $output;
```

### Expected Output

The XSLT transformation should complete successfully.

The file_get_contents function should render the content of the specified template files.

jQuery AJAX requests should transport data as expected.

### Anything else?

**Actual Behavior:**
The XSLT transformation and template rendering appear to work correctly.
However, jQuery AJAX requests fail to transport any data, resulting in incomplete or broken functionality on the client side.
This issue only occurs in the development environment and does not occur in production.

**Comment by michalsn** [comment]
Since the problem only affects the development environment, I assume the Toolbar is at fault.

Please, make sure that the AJAX request can be recognized correctly by setting the appropriate headers:
https://codeigniter.com/user_guide/general/ajax.html#jquery

Another option is View Collector, which may interfere with the data you're returning. You may try to disable it in `app/Config/Toolbar.php`. Just comment the `Views::class` in the `$collectors` array.

For now, I will remove the bug label. If the above does not work for you, feel free to provide more info.

**Comment by michalsn** [comment]
I don't have any feedback - closing.
